src/gt4sd/frameworks/enzeptional/core.py

```diff
@@ -0,0 +1,887 @@
+        mutated_sequences_range: List[str],
+        visited_sequences: set,
+        intervals: List[Tuple[int, int]],
+        batch_size: Optional[int] = None,
+    ) -> Tuple[Set[Any], List[Dict[str, Any]]]:
+        """evaluate sequences.
+        Args:
+            original_sequence_score: float. Score of the original sequence
+            mutated_sequences_range: list of mutated sequences (just concatenated fragents of sequences)
+            visited_sequences: set of sequences already evaluated in the past optimization steps
+            intervals: list of ranges in the sequence, zero-based.
+            batch_size: number of sequences to evaluated in one round
+        Returns:
+            a list of sequences.
+        """
+        temporary_results: List[Dict[str, Any]] = []
+
+        if not batch_size:
+            batch_size = 8
+
+        lst_mutated_sequences: List[str] = []
+
+        for mutated_fragments in mutated_sequences_range:
+            mutated_sequence = reconstruct_sequence_with_mutation_range(
+                sequence=self.sequence,
+                mutated_sequence_range=mutated_fragments.strip(),
+                intervals=intervals,
+            )
+
+            if mutated_sequence not in visited_sequences:
+                visited_sequences.add(mutated_sequence)
+                lst_mutated_sequences.append(mutated_sequence)
+
+        for i in range(0, len(lst_mutated_sequences), batch_size):
+            temporary_results += [
+                scored_sequence
+                for scored_sequence in self.score_sequences(
+                    lst_mutated_sequences[i : i + batch_size]
+                )
+                if scored_sequence["score"] > original_sequence_score
+            ]
+
+        return visited_sequences, temporary_results
+
+    def selection_crossover(
+        self,
+        sequence_from_intervals: str,
+        tmp_results: List[Dict[str, Any]],
+        intervals: List[Tuple[int, int]],
+        selection_method: str,
+        crossover_method: str,
+        crossover_probability: float = 0.5,
+        top_k_selection: Optional[int] = -1,
+    ) -> List[str]:
+        """Perform Selection and Crossover
+
+        Args:
+            sequence_from_intervals (str): orignial sequence extrcated from interval.
+            tmp_results (List[Dict[str, Any]]): the temporary results
+            intervals (List[Tuple[int, int]]): list of ranges in the sequence, zero-based. The same interval is applied to all sequences
+            selection_method (str): methodology used to selection
+            crossover_method (str): methodology used for crossover
+            crossover_probability (float, optional): Crossover probability. Used in case Uniform crossover is selected. Defaults to 0.5.
+            top_k_selection (Optional[int], optional): Number of samples to select. Defaults to -1.
+
+        Returns:
+            List[str]: New samples for the next round of optimization
+        """
```

Comment:
this docstring style is to compliant and will not the rendered properly in the docs, fix it following the other examples